symphony/bdk/core/client/trace_id.py

```diff
@@ -0,0 +1,86 @@
+"""Module responsible for managing the trace ID sent as X-Trace-Id header and logged under the `trace_id` variable
+"""
+import logging
+import random
+import string
+from contextvars import ContextVar
+
+
+def setup_trace_id_log_record_factory():
+    """Adds the `trace_id` variable to the log records.
+
+    :return: the updated log record factory which adds the `trace_id` variable with the current trace id value
+    """
+    old_factory = logging.getLogRecordFactory()
+
+    def trace_id_log_factory(*args, **kwargs):
+        record = old_factory(*args, **kwargs)
+        record.trace_id = DistributedTracingContext.get_trace_id()
+
+        return record
+
+    logging.setLogRecordFactory(trace_id_log_factory)
+
+
+def add_x_trace_id(func):
+    """Decorator of ApiClient.__call_api function so that it adds the `X-Trace-Id` header
+    with the current trace id value to the HTTP call.
+
+    :param func: the function to be decorated
+    :return: the decorated function
+    """
+
+    async def add_x_trace_id_header(*args, **kwargs):
+        if not DistributedTracingContext.has_trace_id():
+            DistributedTracingContext.set_trace_id()
+
+        args[4]["X-Trace-Id"] = DistributedTracingContext.get_trace_id()
```

Comment:
is the args[4] risky to use? (should we check it is here for instance?)

Reply:
+1 It doesn't seem readble to me. 
I am curious to know how are we planning to use it.
I understand you want to use `add_x_trace_id` as a decorator, how do arguments make sense here ?